packages/customize-widgets/src/components/sidebar-block-editor/sidebar-adapter.js

```diff
@@ -53,12 +48,25 @@ export default class SidebarAdapter {
 		this.setting = setting;
 		this.allSettings = allSettings;
 
+		this.locked = false;
+		this.widgetsMap = new WeakMap();
 		this.subscribers = new Set();
 
+		this.history = [
+			this._getWidgetIds().map( ( widgetId ) =>
+				this.getWidget( widgetId )
+			),
+		];
+		this.historyIndex = 0;
+
 		this._handleSettingChange = this._handleSettingChange.bind( this );
 		this._handleAllSettingsChange = this._handleAllSettingsChange.bind(
 			this
 		);
+		this.canUndo = this.canUndo.bind( this );
+		this.canRedo = this.canRedo.bind( this );
+		this.undo = this.undo.bind( this );
+		this.redo = this.redo.bind( this );
```

Comment:
Binding these shouldn't be necessary. It's only needed when we use a class member function as a callback.

Reply:
It's just easier if any components/hooks want to access these methods without having to memoize/bind them again. We don't need these now though.

```js
const { canUndo, canRedo, undo, redo } = sidebar;
```